index.js

```diff
@@ -131,7 +127,6 @@ module.exports = function(file, opt) {
             // possible leftover worker that has no channel
             // estabilished will throw. Ignore.
             try {
-                worker.send({cmd: 'disconnect'});
```

Comment:
There was exception when sending `disconnect` at this stage, because pipe was already closed, and the msv-report-cluster restarted itself because of the exception.

Reply:
Why would the cluster restart if it's in a try?

Reply:
I really don't have idea, here is the log:
```
Nov 19 17:25:25 dev2_webapi_i-0a3a08fc7e08d9bfd NaN>2019-11-19T15 [debug] 25:25.364Z 127.0.0.1 msv-report[28325]:uncaught process exception will shut down worker 1: initial filter stack is empty!
Nov 19 17:25:25 dev2_webapi_i-0a3a08fc7e08d9bfd msv-report [warning] exception dumped in file: /tmp/exception_7064.dump
Nov 19 17:25:25 dev2_webapi_i-0a3a08fc7e08d9bfd msv-report [info] worker 1 has been disconnected from the cluster
Nov 19 17:25:25 dev2_webapi_i-0a3a08fc7e08d9bfd init [warning]  msv-report main process (28318) terminated with status 1
Nov 19 17:25:25 dev2_webapi_i-0a3a08fc7e08d9bfd msv-report [info] logged exception in rollbar https://rollbar.com/occurrence/uuid/?uuid=bad9a828-cf62-4a0d-e800-e20efa094da0
...
Nov 19 17:25:30 dev2_webapi_i-0a3a08fc7e08d9bfd NaN>2019-11-19T15 [debug] 25:30.404Z 127.0.0.1 msv-report[28373]:uncaught process exception will shut down worker 2: write EPIPE
Nov 19 17:25:30 dev2_webapi_i-0a3a08fc7e08d9bfd msv-report [info] gracefully closing mailers: default
Nov 19 17:25:30 dev2_webapi_i-0a3a08fc7e08d9bfd msv-report [info] worker 2 exiting after uncaught process exception
Nov 19 17:25:30 dev2_webapi_i-0a3a08fc7e08d9bfd msv-report [warning] error disconnecting from cluster: write EPIPE
```

The strange is that with `worker.send` the new process of msv-report is also terminated + the `cluster` and we have only the old process working and nothing kills it.

Reply:
wow, ok.  I guess the send is completely async and it's emitting an error that nothing is listening to which then gets thrown

Reply:
From v4 of node `worker.send` https://nodejs.org/api/cluster.html#cluster_worker_send_message_sendhandle_options_callback supports callback parameter (not required). I might try to wrap a promise in try/catch block, but still thinking that shouldn't happen and we could terminate the new instance because it will receive the message.